src/hooks/useAuth.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useEffect, useState } from 'react';
+import { authHookCb } from '../integrations/stytch';
+import { getStytchJwt } from '../utils/browser';
+import { getAuthClient } from '../utils/init';
+/** These next 2 should maybe be dynamically imported, but having trouble with that */
+import { useIdleTimer } from 'react-idle-timer';
+import throttle from 'lodash.throttle';
+
+const TEN_SECONDS_IN_MILLISECONDS = 1000 * 10;
+const FIVE_MINUTES_IN_MILLISECONDS = 1000 * 60 * 5;
+
+export const useAuth = ({
+  onAuth = () => {},
+  onLogout = () => {
+    // @ts-expect-error TS(2322) FIXME: Type 'string' is not assignable to type '(string |... Remove this comment to see the full error message
+    window.location = window.location.origin;
+  }
+}) => {
+  // Need to use this flag because when doing magic link login the onChange
+  // event doesn't seem to be added early enough to catch the first event which
+  // is the one containing the token. subsequent events do not contain the token
+  const [hasAuthed, setHasAuthed] = useState(false);
+  // Use this render state to force re-evaluation of initState, since initState isn't reactive as-is
+  const [render, setRender] = useState(true);
+  authHookCb.cb = () => setRender((prev) => !prev);
+
+  useEffect(() => {
+    const authClient = getAuthClient();
+    if (!authClient) return;
+
+    const unsubscribe = authClient.session.onChange((newSession: any) => {
+      if (hasAuthed) return;
+      // This onChange is just meant to set the token on initial login. It is
+      // not to cache the token. The token is retrieved directly every time it
+      // is needed by the function getStytchJwt
+      const newToken =
+        newSession?.stytch_session?.session_jwt ?? getStytchJwt();
+      if (!newToken) return;
+      setHasAuthed(true);
+      // cb to fetch resources
+      onAuth();
```

Comment:
I should try to listen here to check if the client revokes the session themselves. if so, I need to call onLogout